tasks/eva_2385/sharding_report.py

```diff
@@ -0,0 +1,122 @@
+import argparse
+import getpass
+from os import path
+
+import pandas
+from ebi_eva_common_pyutils.logger import logging_config
+from pymongo import MongoClient
+
+logger = logging_config.get_logger(__name__)
+logging_config.add_stdout_handler()
+
+
+def read_chunks_info(uri, mongo_password):
+    mongo_client = MongoClient(uri, password=mongo_password)
+    config_db = mongo_client["config"]
+    chunk_collection = config_db["chunks"]
+
+    skip_config_db = {
+        "$match":
+            {
+                "ns": {
+                    "$nin": ["config.system.sessions"],
+                }
+            }
+    }
+    group_by_collection_and_shard = {
+        "$group":
+            {
+                "_id": {"collection": "$ns", "shard": "$shard"},
+                "chunks": {"$sum": 1},
+            }
+    }
+    sort_by_db_collection_shard = {
+        "$sort":
+            {
+                "_id.collection": 1,
+                "_id.shard": 1
+            }
+    }
+
+    pipeline = [
+        skip_config_db,
+        group_by_collection_and_shard,
+        sort_by_db_collection_shard
+    ]
+
+    logger.info("reading chunk info from config db's chunk collection...")
+    aggregated_chunk_info = chunk_collection.aggregate(pipeline, batchSize=0)
+
+    result = {}
+    for agg_chunk in aggregated_chunk_info:
+        ns = agg_chunk["_id"]["collection"]
+        curr_shard = agg_chunk["_id"]["shard"]
+        no_of_chunks = agg_chunk["chunks"]
+
+        if ns in result.keys():
+            result[ns]["shards"][curr_shard]["no_of_chunks"] = no_of_chunks
+        else:
+            database = ns.split(".")[0]
+            collection = ns.split(".")[1]
+
+            logger.info(f"reading collection info for db({database}) and collection({collection})")
+            collection_info = mongo_client[database].command("collstats", collection)
+
+            collection_size = int(collection_info["size"])
+            result[ns] = {
+                "database": database,
+                "collection": collection,
+                "collection_size": collection_size,
```

Comment:
Could you please convert the collection sizes to GB for human readability?

Reply:
I had actually started with that, but for dbs with very small size, it was showing as 0, so I reverted it. 
If showing 0 for small db are fine, we can change this or if you want we can show size in both Bytes and GB by adding another column for GB in the existing structure.

Reply:
ok, that sounds good!

Reply:
added size in GB for collection